### PR TITLE
Add doc for kafka transaction

### PR DESCRIPTION
**Pull request by zhfeng**

### CODE REVIEW

File: components/camel-kafka/src/main/docs/kafka-component.adoc [hunk 1]
```diff
@@ -423,5 +423,16 @@ from("kafka:my_topic?headerFilterStrategy=#myStrategy")
 
 `myStrategy` object should be subclass of `HeaderFilterStrategy` and must be placed in the Camel registry, either manually or by registration as a bean in Spring/Blueprint, as it is `CamelContext` aware.
 
+== Kafka Transaction
+
+It needs to add `transactional.id`, `enable.idempotence` and `retries` in `additional-properties` to enable kafka transaction with the producer.
+[source,java]
+----
+from("direct:transaction")
+.to("kafka:my_topic?additional-properties[transactional.id]=1234additional-properties[enable.idempotence]=trueadditional-properties[retries]=5");
```
Comment (davsclaus) [inline components/camel-kafka/src/main/docs/kafka-component.adoc:432]: You need & to separate the options

File: components/camel-kafka/src/main/docs/kafka-component.adoc [hunk 2]
```diff
@@ -423,5 +423,16 @@ from("kafka:my_topic?headerFilterStrategy=#myStrategy")
 
 `myStrategy` object should be subclass of `HeaderFilterStrategy` and must be placed in the Camel registry, either manually or by registration as a bean in Spring/Blueprint, as it is `CamelContext` aware.
 
+== Kafka Transaction
+
+It needs to add `transactional.id`, `enable.idempotence` and `retries` in `additional-properties` to enable kafka transaction with the producer.
```
Comment (davsclaus) [inline components/camel-kafka/src/main/docs/kafka-component.adoc:428]: It needs -> You need

Comment (zhfeng) [reply components/camel-kafka/src/main/docs/kafka-component.adoc]: Done

**Review by davsclaus — APPROVED** [review]

**Comment by davsclaus** [comment]
I think it would be good if we could make configuring the TX options as new endpoint options so its easier and more tooling friendly. Additional properties is only intended for special and rare use-cases.

**Comment by zhfeng** [comment]
Thanks @davsclaus 

- `transactional.id`->`addtional-properties[transactional.id]`
- `idempotence`->`additional-properties[enable.idempotence]`
- `retries`->`additional-properties[retries]`

Does it make sense?

**Comment by davsclaus** [comment]
Yeah but name it transactionId without the dot in the uri option

**Pull request merged**